easy-connect.h

```diff
@@ -1,53 +1,54 @@
+/*
+ * FILE: easy-connect.h
+ *
+ * Copyright (c) 2015 - 2017 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 #ifndef __EASY_CONNECT_H__
 #define __EASY_CONNECT_H__
 
 #include "mbed.h"
 
 #define ETHERNET          1
-#define WIFI_ESP8266      2
-#define MESH_LOWPAN_ND    3
-#define MESH_THREAD       4
-#define WIFI_ODIN         5
-#define WIFI_RTW          6
-#define CELLULAR_ONBOARD  7
-#define WIFI_IDW0XX1      8
+#define WIFI_ESP8266      11
+#define WIFI_ODIN         12
+#define WIFI_RTW          13
+#define WIFI_IDW0XX1      14
+#define MESH_LOWPAN_ND    101
+#define MESH_THREAD       102
+#define CELLULAR_ONBOARD  201
 
-#if MBED_CONF_APP_NETWORK_INTERFACE == WIFI_ESP8266
-#include "ESP8266Interface.h"
+/* Define supersets for WiFi and Mesh */
 
-#ifdef MBED_CONF_APP_ESP8266_DEBUG
-ESP8266Interface wifi(MBED_CONF_APP_ESP8266_TX, MBED_CONF_APP_ESP8266_RX, MBED_CONF_APP_ESP8266_DEBUG);
-#else
-ESP8266Interface wifi(MBED_CONF_APP_ESP8266_TX, MBED_CONF_APP_ESP8266_RX);
-#endif
+#if MBED_CONF_APP_NETWORK_INTERFACE == WIFI_ESP8266
+#define WIFI
 
 #elif MBED_CONF_APP_NETWORK_INTERFACE == WIFI_ODIN
-#include "OdinWiFiInterface.h"
+#define WIFI
```

Comment:
This macro seems too generic (same with `MESH`). Can we prefix this with `EASY_CONNECT_` or something to avoid collisions in user applications?

Reply:
Yes, we can do that (good point).

Reply:
Fixed that now.